src/api/ledger/parse/utils.js

```diff
@@ -4,6 +4,15 @@ const _ = require('lodash');
 const transactionParser = require('ripple-lib-transactionparser');
 const toTimestamp = require('../../../core/utils').toTimestamp;
 const utils = require('../utils');
+const BigNumber = require('bignumber.js');
+
+function adjustQualityForXRP(quality: string, takerGetsCurrency: string,
+    takerPaysCurrency: string) {
+  const shift = (takerGetsCurrency === 'XRP' ? 6 : 0)
```

Comment:
You can do a google search for "order book"s and you'll see that 100% of them are expressed as quantity/price. I've never seen one use "taker gets" and "taker pays" besides in Ripple. Here's an example: https://www.bitstamp.net/market/order_book/

There is still a question about what to do with the redundancy of the flipped book. Traditional financial markets just have conventions to not use one of the two books.